lib/mix_unused/analyzers/unreachable/usages.ex

```diff
@@ -45,12 +46,9 @@ defmodule MixUnused.Analyzers.Unreachable.Usages do
     Debug.debug(modules, &debug/1)
   end
 
-  @spec declared_usages([module() | mfa()], Exports.t()) :: [mfa()]
+  @spec declared_usages([Filter.pattern()], Exports.t()) :: [mfa()]
   defp declared_usages(hints, exports) do
-    Enum.flat_map(hints, fn
-      {m, f, a} -> [{m, f, a}]
-      m -> for {^m, f, a} <- Map.keys(exports), do: {m, f, a}
-    end)
+    Filter.filter_matching(exports, hints) |> Enum.map(fn {mfa, _} -> mfa end)
```

Comment:
Forgot a last bit... here `filter_matching` is returning a map, and we just want its keys, so:
```
exports |> Filter.filter_matching(hints) |> Map.keys()
```

Ah please rename `hints` in `patterns`.